### ISSUE TITLE
Config settings should be added to the blockchain during genesis.

### ISSUE THREAD

**Issue by Sriep** [0chain/0chain#687]
The contents of the config files `0chain.yaml` and `sc.yaml` should be transferred to the  MPT during chain genesis. Currently, this is done during the first smart-contract that touches the settings.

We need these to be the same across the network when settings are added to the MPT. If there is a problem this is best discovered during genesis rather than at a later date.

**Comment by Sriep** [comment]
We should split [0chian.yaml](https://github.com/0chain/0chain/blob/staging/docker.local/config/0chain.yaml) into ~two~ three files.
1. `0chain.yaml:` Contains settings that are stored on the chain. During chain genesis, these settings are added to the MPT. Therafter the 0chain.yaml file is unused. If the 0chain.yaml files do not match across the network during genesis than the chain does not start.

```yaml
development:
  state: true
  dkg: true
  view_change: false
  block_rewards: false
  smart_contract:
    storage: true
    faucet: true
    zrc20: true
    interest: true
    miner: true
    multisig: true
    vesting: true
    zcn: true
  txn_generation:
    wallets: 50
    max_transactions: 0
    max_txn_fee: 10000
    min_txn_fee: 0
    max_txn_value: 10000000000
    min_txn_value: 100
  faucet:
    refill_amount: 1000000000000000

zerochain:
  id: "0afc093ffb509f059c55478bc1a60351cef7b4e9c008a53a6cc8241ca8617dfe"
  decimals: 10
  genesis_block:
    id: "ed79cae70d439c11258236da1dfa6fc550f7cc569768304623e8fbd7d70efae4"

server_chain:
  id: "0afc093ffb509f059c55478bc1a60351cef7b4e9c008a53a6cc8241ca8617dfe"
  owner: "edb90b850f2e7e7cbd0a1fa370fdcc5cd378ffbec95363a7bc0e5a98b8ba5759"
  decimals: 10
  tokens: 200000000
  genesis_block:
    id: "ed79cae70d439c11258236da1dfa6fc550f7cc569768304623e8fbd7d70efae4"
  block:
    min_block_size: 1
    max_block_size: 10
    max_byte_size: 1638400
    min_generators: 2
    generators_percent: 0.2
    replicators: 0
    generation:
      timeout: 15
      retry_wait_time: 5 #millisecondshttps://github.com/0chain/0chain/issues/687
    proposal:
      max_wait_time: 180ms
      wait_mode: static # static or dynamic
    consensus:
      threshold_by_count: 66 # percentage (registration)
      threshold_by_stake: 0 # percent
    sharding:
      min_active_sharders: 25 # percentage
      min_active_replicators: 25 # percentageRF
    validation:
      batch_size: 1000
    reuse_txns: false
    storage:
      provider: blockstore.FSBlockStore # blockstore.FSBlockStore or blockstore.BlockDBStore
  round_range: 10000000
  round_timeouts:
    softto_min: 1500 # in miliseconds
    softto_mult: 1 # multiples of mean network time (mnt)  softto = max{softo_min, softto_mult * mnt}
    round_restart_mult: 10 # number of soft timeouts before round is restarted
    timeout_cap: 0 # 0 indicates no cap
    vrfs_timeout_mismatch_tolerance: 5
  transaction:
    payload:
      max_size: 98304 # bytes
    timeout: 30 # seconds
    min_fee: 0
  client:
    signature_scheme: bls0chain # ed25519 or bls0chain
    discover: true
  messages:
    verification_tickets_to: all_miners # generator or all_miners
  state:
    prune_below_count: 100 # rounds
    sync:
      timeout: 10 # seconds
  stuck:
    check_interval: 10 # seconds
    time_threshold: 60 #seconds
  smart_contract:
    setting_update_period: 200 #rounds
    timeout: 8000ms
  health_check:
    show_counters: true
    deep_scan:
      enabled: false
      settle_secs: 30s
      window: 0 #Full scan till round 0
      repeat_interval_mins: 3m #minutes
      report_status_mins: 1m #minutes
      batch_size: 50
    proximity_scan:
      enabled: true
      settle_secs: 30s
      window: 100000 #number of blocks, Do not make 0 with minio ON, Should be less than minio old block round range
      repeat_interval_mins: 1m #minutes
      report_status_mins: 1m #minutes
      batch_size: 50
```
2. `miner.yaml:` This contains the settings lohttps://github.com/0chain/0chain/issues/687cal to the miner.
```yaml
logging:
  level: "debug"
  console: false # printing log to console is only supported in development mode
  goroutines: false
  memlog: false



  lfb_ticket:
    rebroadcast_timeout: "15s" #
    ahead: 5 # should be >= 5
    fb_fetching_lifetime: "10s" #
  async_blocks_fetching:
    max_simultaneous_from_miners: 100
    max_simultaneous_from_sharders: 30
  dbs:
    events:
      enabled: true
      name: events_db
      user: zchain_user
      password: zchian
      host: postgres #localhost
      port: 5432
      max_idle_conns: 100
      max_open_conns: 200
      conn_max_lifetime: 20s

geolocation:
  latitude: 0
  longitude: 0

delegate_wallet: ""
service_charge: 0.10 # [0; 1) of all fees
number_of_delegates: 10 # max number of delegate pools
min_stake: 0.0 # tokens
max_stake: 100.0 # tokens

minio:
  enabled: false # Enable or disable minio backup, Do not enable with deep scan ON
  worker_frequency: 1800 # In Seconds, The frequency at which the worker should look for files, Ex: 3600 means it will run every 3600 seconds
  num_workers: 5 # Number of workers to run in parallel, Just to make execution faster we can have mutiple workers running simultaneously
  use_ssl: false # Use SSL for connection or not
  old_block_round_range: 250000 # How old the block should be to be considered for moving to cloud, Should be greater than proximity scan window
  delete_local_copy: true # Delete local copy of block once it's moved to cloud

cassandra:
  connection:
    delay: 10 # in seconds
    retries: 10

integration_tests:
  address: host.docker.internal:15210
  lock_interval: 1s
```
Includes fields for geolocation, latitude or longitude.

**Comment by ma2b0043** [comment]
@Sriep when can we expect to close this issue?

**Comment by Sriep** [comment]
https://github.com/0chain/0chain/issues/905

**Comment by peterlimg** [comment]
Initialize config to MPT in genesis block can be a separate task, this is needed for mainnnet but spliting the config files is not a must for mainnet ATM.
